packages/workspace/src/browser/workspace-service.ts

```diff
@@ -9,12 +9,15 @@ import { injectable, inject } from "inversify";
 import URI from "@theia/core/lib/common/uri";
 import { FileSystem, FileStat, FileSystemWatcher } from "@theia/filesystem/lib/common";
 import { WorkspaceServer } from "../common";
+import { FrontendApplication, FrontendApplicationContribution } from '@theia/core/lib/browser';
 
 /**
  * The workspace service.
  */
 @injectable()
-export class WorkspaceService {
+export class WorkspaceService implements FrontendApplicationContribution {
+
+    root: FileStat | undefined;
```

Comment:
I do not understand why do we have a public property here and another public property getter, `tryRoot` too. Perhaps, you can explain this to me. The `tryRoot` at least validates whether the root `FileStat` (if set) is valid or not.

Reply:
it should not be public